packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2278,23 +2274,35 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       leftStartLoc: Position,
       minPrec: number,
     ) {
+      let isSatisfies: ?boolean;
       if (
         tokenOperatorPrecedence(tt._in) > minPrec &&
         !this.hasPrecedingLineBreak() &&
-        this.isContextual(tt._as)
+        (this.isContextual(tt._as) ||
+          (isSatisfies = this.isContextual(tt._satisfies)))
       ) {
-        const node: N.TsAsExpression = this.startNodeAt(
-          leftStartPos,
-          leftStartLoc,
-        );
+        const node: N.TsAsExpression | N.TsSatisfiesExpression =
+          this.startNodeAt(leftStartPos, leftStartLoc);
         node.expression = left;
-        const _const = this.tsTryNextParseConstantContext();
-        if (_const) {
-          node.typeAnnotation = _const;
-        } else {
-          node.typeAnnotation = this.tsNextThenParseType();
-        }
-        this.finishNode(node, "TSAsExpression");
+        node.typeAnnotation = this.tsInType(() => {
+          this.next(); // "as" or "satisfies"
+          if (this.match(tt._const)) {
+            if (isSatisfies) {
+              this.raise(
```

Comment:
I think we don't have to specialize `tt._const` after `tt._satisfies`: `tsParseType` should throw when `const` is seen.